pymeasure/display/widgets/multiplot_widget.py

```diff
@@ -0,0 +1,210 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+import os
+import pyqtgraph as pg
+from functools import partial
+
+from ..curves import ResultsCurve
+from ..Qt import QtCore, QtGui
+from .plot_widget import PlotWidget
+from .plot_frame import PlotFrame
+from .results_dialog import ResultsDialog
+from ...experiment.results import Results
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class MultiPlotWidget(PlotWidget):
+    """ Extends the PlotFrame to allow different columns
+    of the data to be dynamically choosen across multple plots
+    """
+
+    def __init__(self, name, columns, x_axis=None, y_axis=None, num_plots=1, *args, **kwargs):
+
+        super().__init__(name, columns, x_axis, y_axis, setup=False, *args)
+        self.num_plots = num_plots
+        self._setup_ui()
+        self._layout()
```

Comment:
Here, the `setup=False` logic is also not required.

```suggestion
        self.num_plots = num_plots
        super().__init__(name, columns, x_axis=None, y_axis=None, *args)
```

Reply:
Reverted this change